src/test/java/com/ericsson/eiffel/remrem/generate/service/EiffelRemremControllerUnitTest.java

```diff
@@ -0,0 +1,111 @@
+package com.ericsson.eiffel.remrem.generate.service;
+
+
+import java.util.ArrayList;
+import java.util.List;
+
+import java.io.File;
+import java.net.URL;
+import java.util.Scanner;
+
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+
+import org.mockito.Spy;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import com.ericsson.eiffel.remrem.generate.controller.RemremGenerateController;
+import com.ericsson.eiffel.remrem.protocol.MsgService;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonParser;
+
+
+import static org.junit.Assert.assertEquals;
+
+@RunWith(SpringRunner.class)
+public class EiffelRemremControllerUnitTest {
+    
+    @InjectMocks
+    RemremGenerateController unit = new RemremGenerateController();
+    
+    @Mock
+    MsgService service;
+    
+    @Mock
+    MsgService service2;
+    
+
+    @Spy
+    private List<MsgService> msgServices = new ArrayList<MsgService>();
+    static String testInputName = "samplesemantics.json";
```

Comment:
Since we only want to test the logic of RemremGenerateController we do not need a file with input. The input will not be used anyway since we mock the service. The body below could be a very simple JsonElement. Perhaps we could use a mocked body since we do not care about its content.